### ISSUE TITLE
some advise for limit count

### ISSUE THREAD

**Issue by Angry-Rabbit** [openresty/lua-resty-limit-traffic#23]
1. Code line: 54 ~ 69 in method _M.incoming() seemed never run. 
`if not ok then
                if err == "not found" then
                    remaining, err = dict:incr(key, -1, limit)
                    if not remaining then
                        return nil, err
                    end

                    ok, err = dict:expire(key, window)
                    if not ok then
                        return nil, err
                    end

                else
                    return nil, err
                end
end`

2. Can not set limit like req limit module "set_rate", Change cache the list visit times instead of remaining times should be fine for this feature.

**Comment by agentzh** [comment]
@Angry-Rabbit For 1), that bit is for race condition handling, which should be very rare to hit but still possible in the wild. For 2), I don't understand your intention and use case. Will you elaborate?

**Comment by Angry-Rabbit** [comment]
@agentzh Thank you for your answer. I see now. 
For the second point, the scenes is I want to deploy multiple openresty instances as an high availability API gateway server group,  when any of these servers are breakdown or recover, the other servers can automatically adjust the limit control params according to the number of living server nodes detected by the heartbeat.  

**Exmple:**
I've deployed muti server instances, the number of breakdown server instance is **M**, and living server instance is **N**.  and limit rules list like blow: 

```
-- Upper concurrent incoming for each API  limit 500 
local lim_conn, err =   limit_conn.new("my_limit_conn_store", 500/N, 100, 0.5)

-- Limit 5000 count for each ip daily
local lim_count, err = limit_count.new("my_limit_count_store", 5000/N, 3600*24)
```

If some instance breakdown, it can be detected by heartbeat and return the new number of living instance **N2**.  For conn limit, The conn can changed simply like blow:

`limit_conn:set_conn(500/N2)`

But for count limit it seemed cannot do this dynamic change. I think if count limit module provide some functions(like, incoming_amount(), set_limit() ) should be conveniently to do this change, just program like this:

```
-- Get incoming amount
local incoming_amount = limit_count.incoming_amount(key)

-- Dynamic change limit.
-- total_remaining:  The total of remaining amount, stored in cache server, like redis
-- N2:  The number of living server instance
limit_count.set_limit(incoming_amount + total_remainin g/N2)
```

**Comment by agentzh** [comment]
@Angry-Rabbit It's technically possible to adjust the count limit threshold on the fly. Will you contribute a patch for it? Thanks!

**Comment by Angry-Rabbit** [comment]
@agentzh Yes, I'll try. This just in my mind now, need some time to do it.
